pkgs/development/python-modules/yapf/default.nix

```diff
@@ -1,19 +1,56 @@
-{ lib, buildPythonPackage, fetchPypi }:
+{ lib
+, buildPythonApplication
+, fetchFromGitHub
+, nose
+, setuptools
+}:
 
-buildPythonPackage rec {
+buildPythonApplication rec {
   pname = "yapf";
-  version = "0.30.0";
+  version = "0.31.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "3000abee4c28daebad55da6c85f3cd07b8062ce48e2e9943c8da1b9667d48427";
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Yfc1ES16MT6IpqzBD79F1U/d+32O7KKWYm7nfS8LqcQ=";
   };
 
+  nativeBuildInputs = [
+    setuptools
+  ];
+  checkInputs = [
+    nose
+  ];
+
+  checkPhase = ''
+    nosetests
```

Comment:
```suggestion
    runHook preTest

    nosetests

    runHook postTest
```

Reply:
`Test`? Why not `Check`? I am confused.

Reply:
definitely `Check`

Reply:
@SuperSandro2000 Can you clarify why the hooks are important here?

Reply:
Some Hooks expect those phases to be run and overwriting faces can be break those silently.